bus_sdk/messages.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+
+class Message(dict):
+    """
+    Subclass of the built in `dict` type which allows enforcement
+    of required fields.
+    """
+    _type = 'base_msg'
+    required_fields = []
+
+    def __init__(self, *args, **kwargs) -> None:
+        """
+        Initialize as you would a normal `dict`.
+
+        >>> data = {'id': 1234}
+        >>> msg = Message(**data)
+        >>> print(msg)
+        {'id': 1234, 'type': 'base_msg'}"
+
+        # or
+
+        >>> msg = Message(id=1234)
+        >>> print(msg)
+        {'id': 1234, 'type': 'test_msg'}"
+
+        :param args:
+        :param kwargs:
+        """
+        kwargs['type'] = self._type
+        super(Message, self).__init__(*args, **kwargs)
+
+        for field in self.required_fields:
+            if not self.get(field):
+                raise AttributeError('{} is a required field'.format(field))
+
+    def to_json(self):
+        return json.dumps(self)
```

Comment:
Is this needed if it's already a `dict`?

Reply:
I included it because it meant you didn't have to import `json` and pass it through `.dumps()` every time you needed the JSON representation. You can still pass it through `.dumps()` manually if preferred, but it saved me a few lines of code in modules and tests in which I didn't need `json` for anything else other than converting to the `Message` to JSON.